Tests/UnitTests/Networking/Backend/BackendGetVirtualCurrenciesTests.swift

```diff
@@ -0,0 +1,291 @@
+    }
+
+    func testGetVirtualCurrenciesCallsHTTPMethod() {
+        self.httpClient.mock(
+            requestPath: .getVirtualCurrencies(appUserID: Self.userID),
+            response: .init(statusCode: .success, response: Self.twoVirtualCurrenciesResponse)
+        )
+
+        let result = waitUntilValue { completed in
+            self.virtualCurrenciesAPI.getVirtualCurrencies(
+                appUserID: Self.userID,
+                isAppBackgrounded: false,
+                completion: completed
+            )
+        }
+
+        expect(result).to(beSuccess())
+        expect(self.httpClient.calls).to(haveCount(1))
+    }
+
+    // MARK: - Jitterable Delay Tests
+    func testGetVirtualCurrenciesUsesDefaultJitterableDelayWhenAppBackgrounded() {
+        self.httpClient.mock(
+            requestPath: .getVirtualCurrencies(appUserID: Self.userID),
+            response: .init(statusCode: .success, response: Self.twoVirtualCurrenciesResponse)
+        )
+
+        let result = waitUntilValue { completed in
+            self.virtualCurrenciesAPI.getVirtualCurrencies(
+                appUserID: Self.userID,
+                isAppBackgrounded: true,
+                completion: completed
+            )
+        }
+
+        expect(result).to(beSuccess())
+        expect(self.httpClient.calls).to(haveCount(1))
+        expect(self.operationDispatcher.invokedDispatchOnWorkerThreadDelayParam) == JitterableDelay.default
+    }
+
+    func testGetVirtualCurrenciesUsesDefaultJitterableDelayWhenAppNotBackgrounded() {
+        self.httpClient.mock(
+            requestPath: .getVirtualCurrencies(appUserID: Self.userID),
+            response: .init(statusCode: .success, response: Self.twoVirtualCurrenciesResponse)
+        )
+
+        let result = waitUntilValue { completed in
+            self.virtualCurrenciesAPI.getVirtualCurrencies(
+                appUserID: Self.userID,
+                isAppBackgrounded: false,
+                completion: completed
+            )
+        }
+
+        expect(result).to(beSuccess())
+        expect(self.httpClient.calls).to(haveCount(1))
+        expect(self.operationDispatcher.invokedDispatchOnWorkerThreadDelayParam) == JitterableDelay.none
+    }
+
+    // MARK: - Caching Tests
+    func testGetVirtualCurrenciesCachesForSameUserID() {
+        self.httpClient.mock(
+            requestPath: .getVirtualCurrencies(appUserID: Self.userID),
+            response: .init(statusCode: .success,
+                            response: Self.twoVirtualCurrenciesResponse,
+                            delay: .milliseconds(10))
+        )
+        self.virtualCurrenciesAPI.getVirtualCurrencies(
+            appUserID: Self.userID,
+            isAppBackgrounded: false
+        ) { _ in }
+
+        self.virtualCurrenciesAPI.getVirtualCurrencies(
+            appUserID: Self.userID,
+            isAppBackgrounded: false
+        ) { _ in }
+
+        expect(self.httpClient.calls).toEventually(haveCount(1))
```

Comment:
This "eventually", I'm not sure it'd cover against also having count of 2. Perhaps we could add a line that checks that `.toNever(haveCount(2))`?